src/app/server/Mdns.h

```diff
@@ -19,26 +19,34 @@
 
 #include <core/CHIPError.h>
 #include <stddef.h>
+#include <support/BitFlags.h>
 
 namespace chip {
 namespace app {
 namespace Mdns {
 
+enum class KeyValueFlags : uint8_t
+{
+    // indicates whether CM (Commissioning Mode) flag should be set to 1
+    kCommissioningMode = 0x01,
+    // indicates whether AP (Additional Commissioning) flag should be set to 1
+    kAdditionalCommissioning = 0x02,
+};
+
 /// Start operational advertising
 CHIP_ERROR AdvertiseOperational();
 
 /// Set MDNS commissioner advertisement
 CHIP_ERROR AdvertiseCommissioner();
 
 /// Set MDNS commissionable node advertisement
-CHIP_ERROR AdvertiseCommissionableNode();
-
-/// Set MDNS advertisement
-// CHIP_ERROR Advertise(chip::Mdns::CommssionAdvertiseMode mode);
-CHIP_ERROR Advertise(bool commissionableNode);
+CHIP_ERROR AdvertiseCommissionableNode(BitFlags<KeyValueFlags> flags);
 
 /// (Re-)starts the minmdns server
-void StartServer();
+//
+// NOTE: when device has never been commissioned, kAdditionalCommissioning flag will be ignored
+//
+void StartServer(BitFlags<KeyValueFlags> flags);
```

Comment:
Why not simply have StartServer() and immediately follow by Advertise? Why couple StartServer and AdvertiseCommissionableNode ?

Reply:
I think that makes sense but my intention here was just to fix a few bugs in the DNS-SD parameters and not re-visit the current design for method invocation.